inference-engine/src/transformations/src/transformations/common_optimizations/convert_nms_gather_path_to_unsigned.cpp

```diff
@@ -0,0 +1,133 @@
+public:
+    NGRAPH_RTTI_DECLARATION;
+
+    PropagateNMSPath(){
+        MATCHER_SCOPE(PropagateNMSPath);
+
+        auto node_pattern = pattern::wrap_type<
+                opset8::Squeeze,
+                opset8::Unsqueeze,
+                opset8::Reshape,
+                op::util::BroadcastBase,
+                opset8::StridedSlice,
+                opset8::VariadicSplit,
+                opset8::Concat,
+                opset8::Convert>();
+
+        matcher_pass_callback callback = [=](pattern::Matcher &m) {
+            auto node = m.get_match_root();
+            const auto & inputs = node->input_values();
+            if (any_of(inputs.begin(), inputs.end(), [](const Output<Node> & output) {
+                return output.get_node()->get_rt_info().count("NMS_SELECTED_INDICES");
+            })) {
+                auto & rt_info = node->get_rt_info();
+                rt_info["NMS_SELECTED_INDICES"] = make_shared<VariantWrapper<string>>("");
+            }
+            return true;
+        };
+
+        auto m = make_shared<pattern::Matcher>(node_pattern, matcher_name);
+        register_matcher(m, callback);
+    }
+};
+
+NGRAPH_RTTI_DEFINITION(PropagateNMSPath, "PropagateNMSPath", 0);
+
+class UpdateConvertGather: public pass::MatcherPass {
+public:
+    NGRAPH_RTTI_DECLARATION;
+
+    UpdateConvertGather(){
+        MATCHER_SCOPE(UpdateConvertGather);
+
+        auto node_pattern = pattern::wrap_type<op::util::GatherBase>();
+
+        matcher_pass_callback callback = [=](pattern::Matcher &m) {
+            auto gather = dynamic_pointer_cast<op::util::GatherBase>(m.get_match_root());
+            if (!gather)
+                return false;
+
+            auto indices = gather->input_value(1);
+
+            const auto& rt_info = indices.get_node()->get_rt_info();
+            if (!rt_info.count("NMS_SELECTED_INDICES"))
+                return false;
+
+            auto out_type = (indices.get_element_type() == element::i64 ?  element::u64 : element::u32);
+            auto existing_convert = dynamic_pointer_cast<opset8::Convert>(indices.get_node_shared_ptr());
+            if (existing_convert && indices.get_target_inputs().size() == 1) {
+                existing_convert->set_convert_element_type(out_type);
+                existing_convert->validate_and_infer_types();
+            } else {
+                auto new_convert_to_unsigned = make_shared<opset8::Convert>(indices, out_type);
+                gather->input(1).replace_source_output(new_convert_to_unsigned);
+                copy_runtime_info(gather, new_convert_to_unsigned);
+            }
+            return true;
+        };
+
+        auto m = make_shared<pattern::Matcher>(node_pattern, matcher_name);
+        register_matcher(m, callback);
+    }
+};
+
+NGRAPH_RTTI_DEFINITION(UpdateConvertGather, "UpdateConvertGather", 0);
+
+pass::ConvertNmsGatherPathToUnsigned::ConvertNmsGatherPathToUnsigned() {
+    MATCHER_SCOPE(ConvertToUnsignedNmsGather);
```

Comment:
this is not matcher scope. so please remove this macro.

Reply:
done